### PR TITLE
Refactoring sprint2

### PR DESCRIPTION
**Pull request by Eriken79**
changed the schema and refactored user.js and user-services.js

### CODE REVIEW

File: bbbackend/models/user-services.js [hunk 1]
```diff
@@ -32,11 +35,15 @@ mongoose
   )
   .catch((error) => console.log(error));
 
+// function to get users from the MongoDB:
```
Comment (hannahtobiason) [inline bbbackend/models/user-services.js:38]: Thanks for the comments so we know what is going on here, very helpful.

File: bbbackend/models/user-services.js [hunk 2]
```diff
@@ -45,63 +52,73 @@ async function getUsers(username, password) {
...
   }
 }
-function setdefaults(userToAdd) {
-  userToAdd.lists = { listcount: 0 };
-  userToAdd.friends = { friendcount: 0 };
-  userToAdd.groups = { testcount: 0 };
+
+
+// called in the addUser function, gives the new user an empty list for friends, groups, inventory, history
+
```
Comment (hannahtobiason) [inline bbbackend/models/user-services.js:71]: the itemList attribute of a user is going to be a List of json objects of type "item" that each have the attributes itemName, price, quantity, and user. I believe you already updated the schema to reflect that, but we may have neglected the quantity attribute.

File: bbbackend/models/user.js [hunk 3]
```diff
@@ -19,11 +26,19 @@ const UserSchema = new mongoose.Schema(
...
     },
+    groups: {
+      type: Object,
+      required: false,
+    },
     friends: {
       type: Object,
       required: false,
     },
-    lists: {
```
Comment (hannahtobiason) [inline bbbackend/models/user.js:26]: See above comment. Make sure the itemList reflects our schema to include item objects.

**Review by hannahtobiason — COMMENTED** [review]
Great work! the refactoring really helps make it clear what code we actually are using and what it does.

**Comment by MohamedCheour98** [comment]
Great Job Erik.

We are now able to include itemlist as an attribute to every user object. This will help with future development when implement the add item feature. Thanks Erik.

**Pull request merged**